charms/istio-pilot/src/charm.py

```diff
@@ -582,6 +625,18 @@ def _patch_istio_validating_webhook(self):
         )
         self.log.info("istiod-default-validator webhook successfully patched")
 
+    def _use_https(self):
+        if _xor(self.model.config["ssl-crt"], self.model.config["ssl-key"]):
+            # Fail if ssl is only partly configured as this is probably a mistake
+            raise ErrorWithStatus(
+                "Charm config for ssl-crt and ssl-key must either both be set or unset",
```

Comment:
sounds strange
xor does  either `(True & False) or (False & True)`

in message we say that it should be (True&True) or (False&False)